src/AVFoundation/AVAudioSession.cs

```diff
@@ -92,12 +94,13 @@ internal NSString CategoryToToken (AVAudioSessionCategory category)
 			return null;
 		}
 		
-		public NSError SetCategory (AVAudioSessionCategory category)
+		public NSError? SetCategory (AVAudioSessionCategory category)
 		{
-			return SetCategory (CategoryToToken (category));
+			var token = CategoryToToken (category);
+			return token is not null ? SetCategory (token) : null;
 		}
 
-		public NSError SetCategory (AVAudioSessionCategory category, AVAudioSessionCategoryOptions options)
+		public NSError? SetCategory (AVAudioSessionCategory category, AVAudioSessionCategoryOptions options)
 		{
 			NSError error;
 			if (SetCategory (CategoryToToken (category), options, out error))
```

Comment:
If `CategoryToToken` returns a nullable value, the first argument to SetCategory must be nullable, but it isn't, so how does this compile without a warning?

Reply:
@rolfbjarne that is a great question... Yeah this is the generated method signature:
`public virtual bool SetCategory (string category, AVAudioSessionCategoryOptions options, out NSError outError)`
And CategoryToToken says it returns an `NSString?`
Although there are no compiler warnings/errors regarding this..

Reply:
Actually your comment made me come up with a theory :) 

There's an implicit NSString -> string operator: 

https://github.com/xamarin/xamarin-macios/blob/1273190e5fdcea446ce4664031048961a3235e73/src/Foundation/NSString.cs#L158

and this method doesn't have nullability information (yet), which means that the compiler stops complaining about nullability problems.

I think that both here and the previous call to `SetCategory (CategoryToToken (category))`, the best is to tell the compiler to be silent: `SetCategory (CategoryToToken (category)!)` - which is also the current behavior (which is good for this PR).